server.js

```diff
@@ -21,3 +24,57 @@ app.get('/bikes', function(req, res) {
 app.listen(3000, function () {
   console.log('Express challenge app listening on port 3000, check http://localhost:3000');
 });
+
+//challenge1
+app.get('/bikes/reqbyid/:id',function(req, res){
+		
+	var i=0;
+	var returnbike; 
+	while(i< bikes.length){
+	if(bikes[i].id == req.params.id){
+        	returnbike = bikes[i];
+      		return res.status(200).json(returnbike);
+        }
+	i++;
+	}
+  if (i>= bikes.length) return res.status(404).send('The bike you"re looking for was not found');
+});
+//challenge2
+app.get('/bikes/:country',function(req, res) {
+	var returnbikes = []; //array of bikes to return as json
+	for (var i=0;i<bikes.length;i++){
+	if (bikes[i].country === req.params.country) returnbikes.push(bikes[i]);
+	}
+	return res.status(200).json(returnbikes);
+});
+//challenge3
+app.delete('/bikes/deletebike/:id',function(req, res){
+		
+	var i = 0; 
+	while (i< bikes.length){
+	if(bikes[i].id == req.params.id){
+		bikes.splice(i ,1);
```

Comment:
Even if it works, modifying an array in the while is dangerous since you loop over the array length